### PR TITLE
Chapter 9: Advanced login

### PR DESCRIPTION
**Pull request by bangpv97**
Screenshot from 2019-05-29 16-56-49

### CODE REVIEW

**Comment by bangpv97** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -17,4 +18,22 @@ def self.digest string
...
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.blank?
```
Comment (VuThiTranVan) [inline app/models/user.rb]: `if remember_digest.blank?` -> `unless remember_digest`

**Comment by VuThiTranVan** [comment]
merged
1 comment nhỏ, fix rồi mợt đi hì

**Comment by bangpv97** [comment]
merged

**Pull request merged**